plugin/helpers_bots.go

```diff
@@ -72,3 +113,67 @@ func (p *HelpersImpl) EnsureBot(bot *model.Bot) (retBotId string, retErr error)
 
 	return createdBot.UserId, nil
 }
+
+func (p *HelpersImpl) ShouldProcessMessage(post *model.Post, botUserId string, options ...ShouldProcessMessageOption) bool {
+	messageProcessOptions := initialiseOptions(options)
+
+	if !messageProcessOptions.AllowSelf {
+		if post.UserId == botUserId {
+			return false
+		}
+	}
+
+	if !messageProcessOptions.AllowSystemMessages {
+		if post.IsSystemMessage() {
+			return false
+		}
+	}
+
+	if !contains(messageProcessOptions.FilterChannelIDs, post.ChannelId) {
+		channel, _ := p.API.GetChannel(post.ChannelId)
+
+		if !IsBotDMChannel(channel, botUserId) {
```

Comment:
This seams like a special case to me. Currently you skip every message that is not a DM to the bot by default. That's a  bit to restrictive. Maybe add a new `LimitToDM(userID string)` method? Or just rely on the plugin using `messageProcessOptions.FilterChannelIDs`?

Reply:
I think we can use `messageProcessOptions.FilterChannelIDs`. What do you think? The only downside the user of the helper method will have to keep track of all the channelIDs.

Reply:
Just the decision on this is pending. Want to know your opinion.